tests/ops/op_math/test_adjoint_op.py

```diff
@@ -308,6 +308,17 @@ class DummyOp(qml.operation.Operator):
         op = Adjoint(DummyOp(0))
         assert op.is_hermitian == value
 
+    def test_batching_properties(self):
+        """Test the batching properties and methods."""
+
+        base = qml.RX(np.array([1.2, 2.3, 3.4]), 0)
+        op = Adjoint(base)
+        assert op.batch_size == 3
+        assert op.ndim_params == (0,)
+
+        op._check_batching([np.array([1.2, 2.3])])
+        assert op.batch_size == base.batch_size == 2
```

Comment:
This behaviour makes sense if we see `Adjoint(op)` to "own" `op`. :+1: 
However, I'm not aware of any scenario where `Adjoint._check_batching` is called, so this test might be somewhat in void.

Reply:
Updating the data of `Adjoint`?  
```python
@qml.qnode(qml.device('default.qubit', wires=1), diff_method="parameter-shift")
def circuit(x):
    qml.adjoint(qml.RX(x, wires=0))
    return qml.expval(qml.PauliZ(0))

qml.grad(circuit)(qml.numpy.array(0.1))
```
Gets called a bunch.

Reply:
Yeah, this is the result of me missing the updating above :) Thanks!